optimizely/event/event_processor.py

```diff
@@ -0,0 +1,208 @@
+# Copyright 2019 Optimizely
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import threading
+import time
+from datetime import timedelta
+from six.moves import queue
+
+from .entity.user_event import UserEvent
+from .event_factory import EventFactory
+
+
+class EventProcessor(object):
+  """ Class encapsulating event_processor functionality. Override with your own processor
+  providing process method. """
+
+  @staticmethod
```

Comment:
This is the convention being used in Python SDK to provide interface like functionality because python does not have interface structures of its own. The same convention can be found in error_handler.py.
We could use ABC module here and declare it as an @abstractmethod.